sdk/spring/spring-cloud-azure-feature-management/README.md

```diff
@@ -192,10 +192,155 @@ feature-management:
         -
          name: TimeWindowFilter
           parameters:
-            time-window-filter-setting-start: "Wed, 01 May 2019 13:59:59 GMT",
-            time-window-filter-setting-end: "Mon, 01 July 2019 00:00:00 GMT"
+            start: "Wed, 01 May 2019 13:59:59 GMT",
+            end: "Mon, 01 July 2019 00:00:00 GMT"
 ```
 
+The time window can be configured to recur periodically. This can be useful for the scenarios where one may need to turn on a feature during a low or high traffic period of a day or certain days of a week. To expand the individual time window to recurring time windows, the recurrence rule should be specified in the `recurrence` parameter.
+
+**Note:** `start` and `end` must be both specified to enable `recurrence`.
+
+```yaml
+feature-management:
+  feature-flags:
+    feature-v:
+      enabled-for:
+        -
+         name: TimeWindowFilter
+          parameters:
+            start: "Fri, 22 Mar 2024 20:00:00 GMT",
+            end: "Sat, 23 Mar 2024 02:00:00 GMT",
+            recurrence:
+              pattern:
+                type: "Daily",
+                interval: 1
+              range:
+                  type: "Numbered"
+                  numberOfOccurrences: 3
+```
+
+The `recurrence` settings is made up of two parts: `pattern` (how often the time window will repeat) and `range` (for how long the recurrence pattern will repeat). To create a recurrence rule, you must specify both `pattern` and `range`. Any pattern type can work with any range type.
+
+**Advanced:** The time zone offset of the `start` property will apply to the recurrence settings.
+
+#### Recurrence Pattern
+
+There are two possible recurrence pattern types: `Daily` and `Weekly`. For example, a time window could repeat "every day", "every 3 days", "every Monday" or "every other Friday".
+
+- Parameters
+
+  | Property | Relevance | Description                                                                                                                                                                                                                                                               |
+      |-----------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+  | **type** | Required  | Two valid values: `Daily`, `Weekly`.                                                                                                                                                                                                                                      |
```

Comment:
Actually, this `type` can be optional, because we have a default value as `Daily`. 
My concern is that if I announce all parameters to be optional, then maybe users will not define any parameters in recurrence pattern like the following.
```
start: "Mon, 13 May 2024 02:00:00 GMT",
end: "Mon, 13 May 2024 03:00:00 GMT",
  recurrence:
        range:
          type: "NoEnd"
```
Shall we support this? Or throw invalid exception since there's no `pattern` parameter? @mrm9084